Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -794,3 +819,36 @@ open class ComposerVC: _ViewController,
         }
     }
 }
+
+/// searchUsers does an autocomplete search on a list of ChatUser and returns users with `id` or `name` containing the search string
+/// results are returned sorted by their edit distance from the searched string
+/// distance is calculated using the levenshtein algorithm
+/// both search and name strings are normalized (lowercased and by replacing diacritics)
+func searchUsers(_ users: [ChatUser], by searchInput: String, excludingId: String? = nil) -> [ChatUser] {
+    let normalize: (String) -> String = {
+        $0.lowercased().folding(options: .diacriticInsensitive, locale: .current)
+    }
+
+    let searchInput = normalize(searchInput)
+
+    let matchingUsers = users.filter { $0.id != excludingId }
+        .filter { $0.id.contains(searchInput) || (normalize($0.name ?? "").contains(searchInput)) }
+
+    let uniqueUsers = matchingUsers.reduce(into: [String: ChatUser]()) {
+        $0[$1.id] = $1
+    }
+
+    let distance: (ChatUser) -> Int = {
+        min($0.id.levenshtein(searchInput), $0.name?.levenshtein(searchInput) ?? 1000)
+    }
+
+    return uniqueUsers.values.map { $0 }.sorted {
+        distance($0) < distance($1)
+    }
+}
+
+extension String {
+    subscript(index: Int) -> Character {
+        self[self.index(startIndex, offsetBy: index)]
+    }
+}
```

Comment:
It is not really common to use global functions in swift for this kind of stuff. Better to put this `searchUsers()` as a function of the `ComposerVC`, and maybe even put it ´open` in case the customer wants to change the algorithm for some reason.